routers/web/repo/setting_protected_branch.go

```diff
@@ -165,12 +165,25 @@ func SettingsProtectedBranchPost(ctx *context.Context) {
 		return
 	}
 
+	// FIXME: If a new ProtectBranch has a duplicate RuleName, an error should be returned.
+
+	// Currently, if a new ProtectBranch with the same name is created,
+	// the existing ProtectBranch will be updated.
+	// But we cannot modify this logic now because many unit tests rely on it.
+
```

Comment:
I think it should be:

- When creating a new ProtectBranch with the same name, the RuleID should be 0, so it's fine to update the existing one. (Update an existing ProtectBranch by name)
- If the RuleID isn't 0 and the RuleName is duplicate, it should return an error. (Update the name of en existing ProtectBranch to a duplicate value)